src/ir_def/check_ir.ml

```diff
@@ -528,9 +528,9 @@ let rec check_exp env (exp:Ir.exp) : unit =
       check (Show.can_show ot) "show is not defined for operand type";
       typ exp1 <: ot;
       T.text <: t
-    | SerializePrim ots, [exp1] ->
+    | SerializePrim ots, es ->
       check (T.shared (T.seq ots)) "debug_serialize is not defined for operand type";
-      typ exp1 <: T.seq ots;
+      List.iter2 (fun t e -> typ e <: t) ots es;
```

Comment:
@crusso FYI -- this is how the change to use a literal tuple in the ASTs is playing out at the moment.  In particular, I found myself generalizing this IR typing rule to permit any number of arguments.  If this change (or any other one) feels like the wrong direction, please let me know.  Seems okay to me, but I'm using your explanation of the backend code still, without knowing firsthand how it works yet.

Reply:
cc @ggreif

Reply:
Instead of rewriting the implementations of SerializePrim, which might actually be fraught with peril, would it not be easier to just picky-back off the existing code and  
desugar the to_candid arguments sequence to a single argument with an explicit tuple (perhaps eliding the tuple for a unary argument (if necessary)).

Reply:
Finally trying now.

Reply:
So after reverting this check back to how it was, I recall why I changed it, to avoid this:

```
internals:420.24-420.58: IR type error [M0000], PrimE (SerializePrim)
 does not work with 0 arguments
Raised at Ir_def__Check_ir.error.(fun) in file "ir_def/check_ir.ml", line 95, characters 30-92
```

This comes quickly after building the compiler, after it's run on the `base` library.  I'm still not that familiar with all of the testing and making sub-steps, so I'm not certain how to sidestep this yet best, other than to simply generalize the line to handle the empty list.